kernel/src/process/signal/pause.rs

```diff
@@ -206,13 +206,21 @@ impl Pause for WaitQueue {
     }
 }
 
-/// Executes a closure while temporarily blocking some signals for the current POSIX thread.
-pub fn with_signal_blocked<R>(ctx: &Context, mask: SigMask, operate: impl FnOnce() -> R) -> R {
-    let posix_thread = ctx.posix_thread;
-    let sig_mask = posix_thread.sig_mask();
+/// Executes a closure after temporarily adjusting the signal mask of the current POSIX thread.
+pub fn with_sigmask_changed<R>(
+    ctx: &Context,
+    mask_op: impl FnOnce(SigMask) -> SigMask,
+    operate: impl FnOnce() -> R,
+) -> R {
+    let sig_mask = {
+        let posix_thread = ctx.posix_thread;
+        posix_thread.sig_mask()
+    };
 
     let old_mask = sig_mask.load(Ordering::Relaxed);
-    sig_mask.store(old_mask + mask, Ordering::Relaxed);
+
+    let new_mask = mask_op(old_mask);
```

Comment:
I'm not sure why you define new variables this way. If you remove the definition of `posix_thread` and `new_mask`, shouldn't the code be shorter and just as readable?

```rust
    let sig_mask = ctx.posix_thread.sig_mask();

    let old_mask = sig_mask.load(Ordering::Relaxed);
    sig_mask.store(mask_op(old_mask), Ordering::Relaxed);

    let res = operate();

    sig_mask.store(old_mask, Ordering::Relaxed);

    res
```